### DISCUSSION TITLE
Leverage Hunger Games to crowdsource price addition (from price tag images)

### DISCUSSION THREAD

**Discussion by raphodn** [openfoodfacts/open-prices#523] [General]
### What

What if we allowed any contributor to add prices from images sent over by other users ?

This would only apply to "Price tag" images, the ones taken in supermarkets or shops, where we can clearly see the EAN & the price (and ideally the product, though sometimes they are not stored at their correct place).

The image author should send the location & date. Or make it mandatory to have images with EXIF/Geoloc data ?

### How

- a new dedicated section in Open Prices web app
- a new "challenge" section in https://hunger.openfoodfacts.org

### Why

Long-term vision : 
- price tag images would only be crowdsourced (the "owner" isn't important, it's cumbersome to add these prices, we could put in some OCR/ML to speed up the price & EAN extraction)
- users would keep ownership of prices coming from receipts & GDPR requests

### Related Issues

- #308
- #320
- #568
- https://github.com/openfoodfacts/open-prices/discussions/526

**Comment by alexfauquette** [comment]
In such case, I would recommend to encourage user to tur products such that the product barcode is also visible on the picture.

**Reply by serpico** [reply]
> In such case, I would recommend to encourage user to tur products such that the product barcode is also visible on the picture.

I couldn't make what you meant by "to tur products", to picture products ?

**Reply by raphodn** [reply]
I think to "turn" the products

**Reply by raphodn** [reply]
the only thing is that the above product is not always reliable (could correspond to the price tag on the left or right, or could be displayed there by mistake... doesn't happen often, but i've seen a couple of times)

**Comment by TTalex** [comment]
> What if we allowed any contributor to add prices from images sent over by other users ?

Great idea in my opinion !

This relates to one of the [community challenge](https://github.com/openfoodfacts/open-prices-frontend/issues/934) questions:
> 5. Maybe another type of contribution to the challenge should be proofs, flagged as matching the challenge, letting other contributors enter the prices, or even deduce them with OCR.

[Optional] Maybe this could also be a first use case for a "flagging" feature on proofs / prices. I can imagine flags calling for contribution `Help me fill prices`, `Help me add product` and flags calling for moderation `Unrelated proof`, `Proof unclear`, `Price seems incorrect`, `Proof leaks sensitive information` (this one is tricky). Even without moderation tools, it could be useful for data consumers to filter out these. (Note: this might be more a Nutri-Patrol thing https://github.com/openfoodfacts/open-prices-frontend/issues/448)

> The image author should send the location & date. Or make it mandatory to have images with Geoloc data ?

I guess both, if EXIF data is available maybe we could prefill or even skip location and date field, otherwise we force the user to fill them.

We could also imagine a batch upload process for proofs. Like selecting many images from your phone gallery at once.

**Reply by raphodn** [reply]
> We could also imagine a batch upload process for proofs. Like selecting many images from your phone gallery at once.

Yes and I mentioned the same idea above :)
also wouldn't be too hard to develop (leveraging the existing ProofAddSingle page)

But this is not the issue concerned by this discussion, maybe we should create another discussion : "Speed up the current price addition" :sweat_smile: 
edit : moved here : https://github.com/openfoodfacts/open-prices/discussions/526

**Reply by raphodn** [reply]
> What if we allowed any contributor to add prices from images sent over by other users ?

The 2 choices we seem to have, are : 
- we develop this inside Open Prices web app
- we leverage Hunger Games. Do you think it would be easy to connect with OP API @alexfauquette ?

**Reply by alexfauquette** [reply]
> Do you think it would be easy to connect with OP API

Not sure there is a lot to reuse in hunger game. It will not be a logo selection, or a yes/no question to answer, so it's a new UI to create. For me there is no technical advantage to create that UI in HungerGame or OpenPrices.

The only advantage to put such interface in HG are
- to keep OP web app as light/clean as possible (only focused on the user data)
- to let HG user be aware of OP and encourage them to contribute

**Reply by raphodn** [reply]
I'm definitely in favor of having all the fun and bite-size contributions in HG :)
And re-orient OP to just "browsing prices" & "my price dashboard" (possibly filtered only on receipts & gdpr data)

**Comment by raphodn** [comment]
Moving the messages related to speeding up the current contribution workflow here : https://github.com/openfoodfacts/open-prices/discussions/526
cc @serpico

**Comment by raphodn** [comment]
Image examples

|Location|Image|
|---|---|
|Carrefour City in Grenoble|![image](https://prices.openfoodfacts.org/img/0013/BnOzzgNJBL.webp)|
|Bio c Bon Paris|![image](https://prices.openfoodfacts.org/img/0010/lqGHf3ZcVR.webp)|
|?|![image](https://github.com/user-attachments/assets/a71a5f19-4494-4e11-84db-67190caf0e18)|
